components/world-4/breeding/pets.tsx

```diff
@@ -0,0 +1,522 @@
+    }, [appContext])
+
+    const starSignUnlocked = useMemo(() => {
+        if (breeding) {
+            if (breeding.starSignInfinity) {
+                setStarSignEquipped(true);
+            }
+            return breeding.starSignUnlocked;
+        }
+
+        return false;
+    }, [appContext, breeding])
+
+    const starSignInfinity = useMemo(() => {
+        if (breeding) {
+            if (breeding.starSignInfinity) {
+                setStarSignEquipped(true);
+            }
+            return breeding.starSignInfinity;
+        }
+
+        return false;
+    }, [appContext, breeding])
+
+    return (
+        <Box margin={{top: "small"}}>
+            <Box align="center" direction="row" justify="center" gap="small" margin={{bottom: "small"}}>
+                {["All", "Shiny", "Breedability"].map((tabName, index) => (
+                    <TabButton key={index} isActive={activeTab == tabName} text={tabName} clickHandler={() => { setActiveTab(tabName); }} />
+                ))
+                }
+            </Box>
+            <Box direction="row" gap="medium" margin={{bottom: "small"}} wrap>
+            {
+                starSignUnlocked && (activeTab == "Shiny" || activeTab == "Breedability") &&
+                <Box direction='row' gap='xsmall'>
+                    <CheckBox
+                        checked={starSignEquipped}
+                        label="Breedabilli Equipped"
+                        onChange={(event) => {
+                            setStarSignEquipped(event.target.checked);
+                            if(!event.target.checked) {
+                                setSilkrode(false);
+                            }
+                        }}
+                        disabled={starSignInfinity}
+                    />
+                    <TipDisplay
+                        heading="Breedabilli"
+                        size='medium'
+                        maxWidth='medium'
+                        body={
+                            <Box>
+                                <Text size='small'>Looks like you unlocked the Breedabilli star sign</Text>
+                                {
+                                    starSignInfinity ?
+                                    <Text margin={{top:'xsmall'}} size='small'>You always get the star sign bonus thanks to the Infinite Stars Rift bonus</Text>
+                                    :
+                                    <Text margin={{top:'xsmall'}} size='small'>To avoid character checking for a global page, use this checkbox to consider it equipped or not</Text>
+                                }
+                            </Box>
+                        }
+                    >
+                        <CircleInformation size="small" />
+                    </TipDisplay>
+                </Box>
+            }
+            {
+                starSignUnlocked && (activeTab == "Shiny" || activeTab == "Breedability") &&
```

Comment:
any reason why you broke this down to 2 different sections that repeat the same check?

Otherwise I think combining them into one would make more sense

Reply:
Indeed I don't know why, I guess I copied the checkbox block two times and left it like that, gonna just merge them